tools/BuildDrop.ps1

```diff
@@ -0,0 +1,240 @@
+        $View.GetType().InvokeMember("Close", "InvokeMethod", $null, $View, $null)           
+        $MSIDatabase = $null
+        $View = $null
+ 
+        # Return the value
+        return $Value
+    } 
+    catch
+    {
+        Write-Warning -Message $_.Exception.Message ; break
+    }
+}
+
+# ==================================================================================================
+# Getting the ProductCode from the msi
+# ==================================================================================================
+
+Rename-Item "$BuildArtifactsPath\signed\AzurePowerShell.msi" "azure-powershell.$PSVersion.msi"
+
+# Get the ProductCode of the msi
+$msiFile = Get-Item "$BuildArtifactsPath\signed\azure-powershell.$PSVersion.msi"
+$ProductCode = ([string](Get-ProductCode $msiFile)).Trim()
+
+# ==================================================================================================
+# Cloning CodePlex WebPI feed and creating the new branch
+# ==================================================================================================
+
+# Clone your fork of the CodePlex WebPI repository
+$fork = "https://git01.codeplex.com/forks/$CodePlexUsername/$CodePlexFork"
+git clone $fork $CodePlexFork
+
+cd $CodePlexFork
+
+# Create a branch that's in the format of YYYY-MM-DDTHH-MM
+$date = Get-Date -Format u
+$branch = $date.Substring(0, $date.Length - 4).Replace(":", "-").Replace(" ", "T");
+git checkout -b $branch
+
+# ==================================================================================================
+# Update the DH_AzurePS.xml file
+# ==================================================================================================
+
+cd "Src\azuresdk\AzurePS"
+
+# Get the text for DH_AzurePS.xml
+$content = Get-Content "DH_AzurePS.xml"
+
+# $newContent will be the text for the updated DH_AzurePS.xml
+$newContentLength = $content.Length + 3
+$newContent = New-Object string[] $newContentLength
+
+$VSFeedSeen = $False
+$PSGetSeen = $False
+$buffer = 0
+
+for ($idx = 0; $idx -lt $content.Length; $idx++)
+{
+    # Flag that we will be looking at the entries for DH_WindowsAzurePowerShellVSFeed next
+    if ($content[$idx] -like "*VSFeed*")
+    {
+        $VSFeedSeen = $True
+    }
+
+    # Flag that we will be looking at the entry for DH_WindowsAzurePowerShellGet next
+    if ($content[$idx] -like "*PowerShellGet*")
+    {
+        $PSGetSeen = $True
+    }
+
+    # Check if we are looking at the DiscoveryHints for DH_WindowsAzurePowerShellVSFeed
+    # and if we have reached the end of the entry so we can add the new msi Product Code
+    if ($VSFeedSeen -and $content[$idx] -like "*</or>*")
+    {
+        $newContent[$idx] =     "      <discoveryHint>"
+        $newContent[$idx + 1] = "        <msiProductCode>$ProductCode</msiProductCode>"
+        $newContent[$idx + 2] = "      </discoveryHint>"        
+
+        # Change the buffer size to include the three lines just added
+        $buffer = 3
```

Comment:
I thought about it, but didn't account for the case where the ordering of the file could change. That is a good idea. I will mark as false here.